stream-chat-android-client/src/main/java/io/getstream/chat/android/client/call/RetryCall.kt

```diff
@@ -41,25 +44,32 @@ internal class RetryCall<T : Any>(
 ) : Call<T> {
 
     private var job: Job? = null
+    private val canceled = AtomicBoolean(false)
 
-    override fun execute(): Result<T> = runBlocking {
-        callRetryService.runAndRetry {
-            originalCall
-        }
-    }
+    override fun execute(): Result<T> = runBlocking { await() }
 
     override fun enqueue(callback: Call.Callback<T>) {
         scope.launch {
-            val result = callRetryService.runAndRetry {
-                originalCall
-            }
+            val result = await()
+            yield()
             withContext(DispatcherProvider.Main) {
                 callback.onResult(result)
             }
         }
     }
```

Comment:
Should we check a cancellation scenario here?
```kotlin
    override fun enqueue(callback: Call.Callback<T>) {
        if (canceled.get()) {
            callback.onResult(Result.error(ChatError("The call was canceled before its execution")))
            return
        }
        scope.launch {
            if (canceled.get()) {
                callback.onResult(Result.error(ChatError("The call was canceled before complete its execution")))
                return@launch
            }
            val result = await()
            yield()
            withContext(DispatcherProvider.Main) {
                if (!canceled.get()) {
                    callback.onResult(result)
                }
            }
        }
    }
```

Reply:
Not sure if we need to omit callback calls on the cancellation.
I guess if users cancel a call they might expect silence from the callback 🤔

Reply:
If it is canceled, we don't want to be notified on the callback.
Imagine the callback is being executed into an activity that is already destroyed, we don't want this code arrives there after we cancel the `Call`

Reply:
About to check the cancelation excenario, it should be done with the `yield()`, but as you catch on previous message, we are not storing the job, so it is not canceled :facepalm: 
Let me fix it